.gitlab-ci.yml

```diff
@@ -49,9 +49,13 @@ stages:
     - echo y | ~/android_sdk/cmdline-tools/latest/bin/sdkmanager --install "$ANDROID_EMULATOR_IMAGE"
     - yes | ~/android_sdk/cmdline-tools/latest/bin/sdkmanager --licenses || true
     - echo "no" | ~/android_sdk/cmdline-tools/latest/bin/avdmanager --verbose create avd --force --name "$EMULATOR_NAME" --package "$ANDROID_EMULATOR_IMAGE"
-  run-instrumented-integration:
+  compile_instrumented_integration:
     - $ANDROID_HOME/emulator/emulator -avd "$EMULATOR_NAME" -grpc-use-jwt -no-snapstorage -no-audio -no-window -no-boot-anim -verbose -qemu -machine virt &
     - GRADLE_OPTS="-Xmx3072m" ./gradlew :instrumented:integration:assembleDebug :instrumented:integration:assembleDebugAndroidTest --stacktrace --no-daemon
+  compile_instrumented_integration_for_api_21:
+    - $ANDROID_HOME/emulator/emulator -avd "$EMULATOR_NAME" -grpc-use-jwt -no-snapstorage -no-audio -no-window -no-boot-anim -verbose -qemu -machine virt &
+    - GRADLE_OPTS="-Xmx3072m" ./gradlew :instrumented:integration:assembleDebug :instrumented:integration:assembleDebugAndroidTest -Puse-api21-java-backport -Puse-desugaring --stacktrace --no-daemon
+  run-instrumented-integration:
```

Comment:
you can still keep `run-instrumented-integration` step without splitting and use the same technique to append `-Puse-api21-java-backport -Puse-desugaring` flags as I used in https://github.com/DataDog/dd-sdk-android/pull/2193

also do we need `use-desugarding` flag? seems we can simply use `use-api21-java-backport`

Reply:
ok will try that technique instead. But regarding desugaring flag...I would rather keep them separated (one is for desugaring, the other is for java backport)